cmd/controller-manager/app/controller-manager.go

```diff
@@ -212,16 +218,98 @@ func getFederationConfig(opts *options.Options) *corev1a1.FederationConfig {
+func getInt(specValue, defaultValue int) int {
+	if specValue == 0 {
+		return defaultValue
+	}
+	return specValue
+}
+
+func setDefaultFederationConfig(fedConfig *corev1a1.FederationConfig) {
+	spec := &fedConfig.Spec
+	if spec.LimitedScope {
+		spec.RegistryNamespace = getString(spec.RegistryNamespace, fedConfig.Namespace)
+	} else {
+		spec.RegistryNamespace = getString(spec.RegistryNamespace, util.MulticlusterPublicNamespace)
+	}
+
+	duration := &spec.ControllerDuration
+	duration.AvailableDelay.Duration = getDuration(duration.AvailableDelay, util.DefaultClusterAvailableDelay)
+	duration.UnavailableDelay.Duration = getDuration(duration.UnavailableDelay, util.DefaultClusterUnavailableDelay)
+
+	election := &spec.LeaderElect
+	election.ResourceLock = getString(election.ResourceLock, util.DefaultLeaderElectionResourceLock)
+	election.RetryPeriod.Duration = getDuration(election.RetryPeriod, util.DefaultLeaderElectionRetryPeriod)
+	election.RenewDeadline.Duration = getDuration(election.RenewDeadline, util.DefaultLeaderElectionRenewDeadline)
+	election.LeaseDuration.Duration = getDuration(election.LeaseDuration, util.DefaultLeaderElectionLeaseDuration)
+
+	healthCheck := &spec.ClusterHealthCheck
+	healthCheck.PeriodSeconds = getInt(healthCheck.PeriodSeconds, util.DefaultClusterHealthCheckPeriod)
+	healthCheck.TimeoutSeconds = getInt(healthCheck.TimeoutSeconds, util.DefaultClusterHealthCheckTimeout)
+	healthCheck.FailureThreshold = getInt(healthCheck.FailureThreshold, util.DefaultClusterHealthCheckFailureThreshold)
+	healthCheck.SuccessThreshold = getInt(healthCheck.SuccessThreshold, util.DefaultClusterHealthCheckSuccessThreshold)
+
+}
+
+func updateFederationConfig(config *rest.Config, fedConfig *corev1a1.FederationConfig) {
+	name := fedConfig.Name
+	namespace := fedConfig.Namespace
+	qualifiedName := util.QualifiedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+
+	configResource := &corev1a1.FederationConfig{}
+	client := genericclient.NewForConfigOrDieWithUserAgent(config, "federationconfig")
+	err := client.Get(context.Background(), configResource, namespace, name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		glog.Fatalf("Error retrieving FederationConfig %q: %v", qualifiedName.String(), err)
+	}
+	if apierrors.IsNotFound(err) {
+		// if `--federation-config` is specifed but there is not FederationConfig resource accordingly
+		err = client.Create(context.Background(), fedConfig)
+		if err != nil {
+			glog.Fatalf("Error creating FederationConfig %q: %v", qualifiedName.String(), err)
+		}
+	} else {
+		configResource.Spec = fedConfig.Spec
+		err = client.Update(context.Background(), configResource)
+		if err != nil {
+			glog.Fatalf("Error updating FederationConfig %q: %v", qualifiedName.String(), err)
+		}
+	}
+}
+
 func setOptionsByFederationConfig(opts *options.Options) {
 	fedConfig := getFederationConfig(opts)
 	if fedConfig == nil {
-		return
+		// Both `--federation-config` flag and `FederationConfig` resource are not used
```

Comment:
-> `FederationConfig` could not be sourced from `--federation-config` or from the API.